x-pack/plugins/apm/server/routes/services/route.ts

```diff
@@ -546,23 +552,21 @@ const serviceThroughputRoute = createApmServerRoute({
       offset,
       start,
       end,
+      documentType,
+      rollupInterval,
```

Comment:
Perhaps co-locate these. Does it ever make sense to pass one and not the other? Feels like they belong together.
```suggestion
      apmDataSource: { documentType, rollupInterval }
```

Reply:
Query parameters need to be flat, and in some cases `bucketSizeInSeconds` should be added as well. The other option is "merge" them in the route, but it will complicate the type a little. E.g. you have to write `ApmDataSource<ApmServiceTransactionDocumentType>`. Not sure if it's worth it but I can make that change if you feel strongly about it.